src/mkdocs_include_markdown_plugin/event.py

```diff
@@ -616,12 +617,12 @@ def found_include_markdown_tag(  # noqa: PLR0912, PLR0915
 
         return text_to_include
 
-    markdown = tags['include'].sub(
-        found_include_tag,
+    markdown = tags['include-markdown'].sub(
+        found_include_markdown_tag,
         markdown,
     )
-    return tags['include-markdown'].sub(
-        found_include_markdown_tag,
+    return tags['include'].sub(
+        found_include_tag,
```

Comment:
Why do you've changed the order of processing?

Reply:
because `recursive` directives are only possible in `include` tags. If i applied the substitution in the the markdown adding a very long line, then on the second pattern search (for `include-markdown`) it would get stuck anyway

Reply:
> then on the second pattern search

Not if you put the exclude before regex sustitutions as you're doing in this PR.

Reply:
are you suggesting that if i kept the order, the following file wouldn't hang?
```markdown
first line
{% include-markdown "some/markdown.md" %}
middle line
{% include "long/lines.html" recursive=false %}
last line
```

Well... It does. Because with the first substitution the markdown text becomes this
```markdown
first line
{% include-markdown "some/markdown.md" %}
middle line
<SUPERLONGLINE>
last line
```

and in this current state you cannot apply a `regex.sub(foo, markdown_text)`

Reply:
Ideally, we should replace all substitutions at once, but with the current implementation is not possible. Tipically text processors declare some strange characters as placeholders to replace them with the new content at once.

If you feel that the whole refactor is too much, don't mind, implement only the `include` directive and I'll open another issue to track this problem.

Reply:
yeah i reckon working with placeholders would be the better approach.
However, i think this would go out of the scope of this PR.

Just mind that if you decide, in the future, to go the route of placeholders, you should first do a full pass that checks whether the placeholder's sequence is already present. If so, escape it, and then proceed with the work!